tests/sklearn/test_sklearn_models.py

```diff
@@ -1139,6 +1162,65 @@ def check_load_fitted_sklearn_linear_models(model_class, n_bits, x, y, check_flo
     )
 
 
+def check_rounding_consistency(
+    model,
+    x,
+    y,
+    predict_method,
+    metric,
+    is_weekly_option,
+):
+    """Test that Concrete ML without and with rounding are 'equivalent'."""
+
+    # Run the test with more samples during weekly CIs
+    if is_weekly_option:
+        fhe_samples = 5
+    else:
+        fhe_samples = 1
+
+    fhe_test = get_random_samples(x, fhe_samples)
+
+    # Check that rounding is enabled
+    rounding_enabled = os.getenv("USE_ROUNDING") == "True"
+    assert rounding_enabled
+
+    # Fit and compile with rounding enabled
+    circuit_with_rounding = fit_and_compile(model, x, y)
+
+    rounded_predict_quantized = predict_method(x, fhe="disable")
+    rounded_predict_simulate = predict_method(x, fhe="simulate")
+    rounded_predict_fhe = predict_method(fhe_test, fhe="execute")
+
+    # We repeat the same experiment, but this time with rounding disabled
+    with pytest.MonkeyPatch.context() as mp_context:
+
+        # Disable rounding
+        mp_context.setenv("USE_ROUNDING", "False")
+
+        # Check that rounding is disabled
+        rounding_disabled = os.environ.get("USE_ROUNDING") == "False"
+        assert rounding_disabled
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", category=DeprecationWarning)
```

Comment:
Better to catch the warning and check that you have the right message than ignoring it imo, especially since you're expecting a precise warning message here

Reply:
Done